TCAT/Controllers/HomeOptionsCardViewController.swift

```diff
@@ -0,0 +1,561 @@
+    func setupInfoButton() {
+        infoButton.imageEdgeInsets = UIEdgeInsets(top: 8, left: 8, bottom: 8, right: 0)
+        infoButton.contentVerticalAlignment = .center
+        infoButton.tintColor = .black
+        infoButton.addTarget(self, action: #selector(openInformationScreen), for: .touchUpInside)
+        view.addSubview(infoButton)
+    }
+    
+    func setupSearchBarSeparator() {
+        searchBarSeparator = UIView()
+        searchBarSeparator.backgroundColor = Colors.backgroundWash
+        view.addSubview(searchBarSeparator)
+    }
+    
+    func setupConstraints() {
+        infoButton.snp.makeConstraints { (make) in
+            make.centerY.equalTo(searchBar)
+            make.trailing.equalToSuperview().inset(16)
+            make.width.equalTo(30)
+            make.height.equalTo(38)
+        }
+        
+        searchBar.snp.makeConstraints { (make) in
+            make.leading.top.equalToSuperview()
+            make.trailing.equalTo(infoButton.snp.leading)
+            make.height.equalTo(searchBarHeight)
+        }
+        
+        searchBarSeparator.snp.makeConstraints { (make) in
+            make.leading.trailing.equalToSuperview()
+            make.top.equalTo(searchBar.snp.bottom)
+            make.height.equalTo(searchBarSeparatorHeight)
+        }
+        
+        tableView.snp.makeConstraints { (make) in
+            make.leading.trailing.bottom.equalToSuperview()
+            make.top.equalTo(searchBarSeparator.snp.bottom)
+        }
+    }
+    
+    func createSections() -> [Section] {
+        var allSections: [Section] = []
+        let recentSearchesSection = Section(type: .recentSearches, items: recentLocations)
+        let seeAllStops = Place(name: Constants.Cells.seeAllStopsIdentifier)
+        let seeAllStopsSection = Section(type: .seeAllStops, items: [seeAllStops])
+        var favoritesSection = Section(type: .favorites, items: favorites)
+        if favoritesSection.items.isEmpty {
+            let addFavorites = Place(
+                name: Constants.General.firstFavorite,
+                placeDescription: Constants.General.tapHere)
+            addFavorites.type = .busStop // Special exception to make pin blue for favorite!
+            favoritesSection = Section(type: .favorites, items: [addFavorites])
+        }
+        allSections.append(favoritesSection)
+        allSections.append(recentSearchesSection)
+        allSections.append(seeAllStopsSection)
+        return allSections.filter { !$0.items.isEmpty || $0.type == .seeAllStops }
+    }
+    
+    func tableViewContentHeight() -> CGFloat {
+        var size: CGFloat = 0
+        for section in sections {
+            switch section.type {
+            case .favorites, .recentSearches: size += headerHeight
+            case .seeAllStops: size += HeaderView.separatorViewHeight
+            default: break
+            }
+            size += tableViewRowHeight*CGFloat(section.items.count)
+        }
+        
+        return size
```

Comment:
you can actually do this whole function in one line doing something like
```
return sections.reduce(...)
```
look at docs here: (https://useyourloaf.com/blog/swift-guide-to-map-filter-reduce/)